### ISSUE TITLE
Invisible distinction between `*Shape` and `*tuple[*Shape]` causes spurious type errors

### ISSUE THREAD

**Issue by colehaus** [microsoft/pyright#7389]
```py
from __future__ import annotations

from typing import Any, Generic, Literal, TypeVar, TypeVarTuple, cast

Shape = TypeVarTuple("Shape")
DType = TypeVar("DType")
Dim1 = TypeVar("Dim1")


class ndarray(Generic[*Shape, DType]):  # noqa: N801
    ...


def insert(
    arr: ndarray[Dim1, *Shape, DType],
    obj: int,
    values: ndarray[*Shape, DType],
    axis: Literal[0],
) -> ndarray[Any, *Shape, DType]:
    raise NotImplementedError


def prepend1(
    arr: ndarray[Dim1, *Shape, DType],
    values: ndarray[*Shape, DType],
) -> ndarray[Any, *Shape, DType]:
    values_: ndarray[*Shape, DType] = values
    return insert(arr, 0, values_, axis=0) # Type error here


def prepend2(
    arr: ndarray[Dim1, *Shape, DType],
    values: ndarray[*Shape, DType],
) -> ndarray[Any, *Shape, DType]:
    values_: ndarray[*Shape, DType] = cast(ndarray[*tuple[*Shape], DType], values)
    return insert(arr, 0, values_, axis=0)
```

```sh
❯ pyright --version
pyright 1.1.352
❯ pyright ml_scratch/type.py
ml_scratch/type.py
  ml_scratch/type.py:28:27 - error: Argument of type "ndarray[*Shape@prepend1, DType@prepend1]" cannot be assigned to parameter "values" of type "ndarray[*Shape@insert, DType@insert]" in function "insert"
    "ml_scratch.type.ndarray" is incompatible with "ml_scratch.type.ndarray"
      Type parameter "Shape@ndarray" is invariant, but "*Shape@prepend1" is not the same as "*tuple[*Shape@prepend1]" (reportArgumentType)
```
As you can see, pyright complains about the first `insert` call in `prepend1` but not the second in `prepend2` even though the declared type of `values_` is the same in both cases.

mypy 1.8.0 complains about neither.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.353, which I just published.
